src/clarity.rs

```diff
@@ -726,115 +1061,199 @@ pub fn invoke_command(invoked_by: &str, args: &[String]) {
-            match run_analysis(&contract_id, &mut ast, &mut analysis_db, true) {
+            match run_analysis_free(&contract_id, &mut ast, &mut analysis_marf, true) {
                 Ok(_) => {
                     let result = vm_env.get_exec_environment(None).eval_raw(&content);
                     match result {
-                        Ok(x) => {
-                            println!("Program executed successfully! Output: \n{}", x);
-                        }
-                        Err(error) => {
-                            eprintln!("Program execution error: \n{}", error);
-                            panic_test!();
-                        }
+                        Ok(x) => (
+                            0,
+                            Some(json!({
+                                "output": serde_json::to_value(&x).unwrap()
+                            })),
+                        ),
+                        Err(error) => (
+                            1,
+                            Some(json!({
+                                "error": {
+                                    "runtime": serde_json::to_value(&format!("{}", error)).unwrap()
+                                }
+                            })),
+                        ),
                     }
                 }
-                Err(error) => {
-                    eprintln!("Type check error.\n{}", error);
-                    panic_test!();
-                }
+                Err((error, _)) => (
+                    1,
+                    Some(json!({
+                        "error": {
+                            "analysis": serde_json::to_value(&format!("{}", error)).unwrap()
+                        }
+                    })),
+                ),
             }
         }
         "eval" => {
-            let evalInput = get_eval_input(invoked_by, args);
-            let vm_filename = if args.len() == 3 { &args[2] } else { &args[3] };
+            let mut argv: Vec<String> = args.into_iter().map(|x| x.clone()).collect();
+
+            let costs = if let Ok(Some(_)) = consume_arg(&mut argv, &["--costs"], false) {
+                true
+            } else {
+                false
+            };
+
+            let evalInput = get_eval_input(invoked_by, &argv);
+            let vm_filename = if argv.len() == 3 { &argv[2] } else { &argv[3] };
+            let header_db =
+                friendly_expect(CLIHeadersDB::resume(vm_filename), "Failed to open CLI DB");
             let marf_kv = friendly_expect(
                 MarfedKV::open(vm_filename, None),
                 "Failed to open VM database.",
             );
-            let header_db = CLIHeadersDB::new(&vm_filename);
-            let result = in_block(vm_filename, marf_kv, |mut marf| {
+            let mainnet = header_db.is_mainnet();
+
+            let (_, _, result_and_cost) = in_block(header_db, marf_kv, |header_db, mut marf| {
                 let result = {
```

Comment:
> You could also just return result_and_cost from this closure because that's all that's being used.

`in_block` needs to return `(CLIHeadersDB, MarfedKV, R)` because there are directives like `initialize` that must invoke `in_block` more than once, but use the same header DB and MARF instance.  Hence the outer `let (_, _, result_and_cost) =` instead of `let result_and_cost =`.